test/api/resolvetask_test.js

```diff
@@ -142,16 +142,158 @@ suite('Report task completed', function() {
       reason:     'malformed-payload'
     });
 
-    var m1 = await gotMessage;
-    assume(m1.payload.status.runs[0].state).equals('exception');
+    var {payload: {status: s1}} = await gotMessage;
+    assume(s1.runs[0].state).equals('exception');
+    assume(s1.runs[0].reasonResolved).equals('malformed-payload');
+
+    debug("### Reporting task exception (again)");
+    await helper.queue.reportException(taskId, 0, {
+      reason:     'malformed-payload'
+    });
+
+    debug("### Check status of task");
+    var {status: s2} = await helper.queue.status(taskId);
+    assume(s2.runs[0].state).equals('exception');
+    assume(s2.runs[0].reasonResolved).equals('malformed-payload');
+  });
+
+  test("reportException (resource-unavailable) is idempotent", async () => {
+    var taskId = slugid.v4();
+    var allowedToFailNow = false;
+    var gotMessage = null;
+
+    await helper.events.listenFor('except', helper.queueEvents.taskException({
```

Comment:
naming the event you're listening for "exception" was just too many characters? :)